### ISSUE TITLE
Nested Navigation Reset not working correctly

### ISSUE THREAD

**Issue by XeniousSoftware** [react-community/react-navigation#1689]
```
   const TabNav = TabNavigator({
      Home: {screen: HomeScreen},
      Favourite: {screen: FavouriteScreen},
      Cart: {screen: CartScreen},
      Account: {screen: AccountScreen},
    },{lazy:true});
    
    const SplashNav=StackNavigator({
          SplashModal:{screen: SplashScreen},
          Login:{screen:LoginScreen},
          Forgot: {screen: ForgotPasswordScreen},
          SearchLocation: {screen:SearchLocationScreen},
          SearchStore: {screen:SearchStoreScreen}
        },{mode:'modal',headerMode:'none'});

    const StackNav=StackNavigator({
      Splash: {screen: SplashNav},
      Tabs: {screen: TabNav},
      Page: {screen: PageScreen},
    });
```
I have 3 navigator like this. so `StackNav` is the nested navigator. when i try to reset navigation from `SplashScreen` to `Tabs` i get following error:

There is no route defined for key Tabs.
Must be one of: 'SplashModal','Login','Forgot','SearchLocation','SearchStore'

so i am not able to reset navigation to Tabs.

but if i switch the position of Splash and Tabs like this

```
    const StackNav=StackNavigator({
      Tabs: {screen: TabNav},
      Splash: {screen: SplashNav},
      Page: {screen: PageScreen},
    });
```
i am able to reset navigation from HomeScreen to Splash. it works perfect!

So basically from a nested StackNavigator i am unable to reset to root screens.

is there anyone who can help?

**Comment by XeniousSoftware** [comment]
You can see the problem here:

https://snack.expo.io/HJ-HsaN-b

**Comment by 0x6e6562** [comment]
Not sure how official this is, but #1384 fixed the issue for me.

**Comment by XeniousSoftware** [comment]
thank you @0x6e6562 you have pointed me into the right direction.

**Comment by yashojha19** [comment]
Have you found the solution of this problem ?
